File: java/candy/src/impl/EatingRequest.java
package impl;

import contracts.Candy;
import impl.producerconsumer.CandyEatingRequestCallback;

/**
 * Запрос на поедание конфеты с возможностью уведомления о завершении поедания.
 */
public class EatingRequest {
    private final Candy candy;
    private final CandyEatingRequestCallback callback;

    public Candy getCandy() {
        return candy;
    }

    public EatingRequest(Candy candy, CandyEatingRequestCallback callback) {
        this.candy = candy;
        this.callback = callback;
    }

    public void complete() throws InterruptedException {
        callback.complete();
    }
}

File: CSharp/CandyEaterFacilityTest/CandyEaterFacilityTest/LockFreeProducerConsumerCandyEatingFacility/EatingRequest.cs
namespace CandyEaterFacilityTest.LockFreeProducerConsumerCandyEatingFacility
{
    public class EatingRequest
    {
        public ICandy Candy { get; private set; }
        public ICallback Callback { get; private set; }

        public EatingRequest(ICandy candy, ICallback callback)
        {
            Candy = candy;
            Callback = callback;
        }
    }
}